### PR TITLE
[EventHub] az eventhubs namespace create : Add Managed Identity parameters to namespace create and update cmdlets

### PR DESCRIPTION
**Pull request by v-Ajnava**
**Description**  
This PR is for add parameters to support for BYOK feature in az eventhubs namespace create and az eventhubs namespace update commands 

**Testing Guide**  
az eventhubs namespace create --resource-group {rg} --name {namespacename} --location {loc} --tags {tags} --sku {sku} --enable-auto-inflate {isautoinflateenabled} --maximum-throughput-units {maximumthroughputunits} --zone-redundant {zoneredundant} --cluster-arm-id {clusterarmid} --enable-identity {enableidentity}

added below optional properties in create namespace command: 
--zone-redundant
--cluster-arm-id
--enable-identity

az eventhubs namespace update --resource-group {rg} --name {namespacename} --tags {tags2} --maximum-throughput-units {maximumthroughputunits_update} --key-source {key_source} --key-properties {key_properties}

added below optional properties in update namespace command: 
--key-source
 --key-properties
--user-identity

**History Notes**  
[eventhubs][][]: az eventhubs namespace create : add managed identity parameters
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
EventHub

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 1]
```diff
@@ -44,6 +44,17 @@ def load_arguments_eh(self, _):
         c.argument('maximum_throughput_units', type=int, help='Upper limit of throughput units when AutoInflate is enabled, vaule should be within 0 to 20 throughput units. ( 0 if AutoInflateEnabled = true)')
         c.argument('default_action', arg_group='networkrule', options_list=['--default-action'], arg_type=get_enum_type(['Allow', 'Deny']),
                    help='Default Action for Network Rule Set.')
+        c.argument('zone_redundant', options_list=['--zone-redundant'], arg_type=get_three_state_flag(),
+                   help='Enabling this property creates a Standard EventHubs Namespace in regions supported availability zones')
+
+    with self.argument_context('eventhubs namespace create') as c:
+        c.argument('cluster_arm_id', options_list=['--cluster-arm-id'], help='luster ARM ID of the Namespace')
+        c.argument('identity', options_list=['--enable-identity'], arg_type=get_three_state_flag(), help='A boolean value that indicates whether Managed Identity is enabled.')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:52]: is it possible to use `--assign-identity` to align with other RPs?

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 2]
```diff
@@ -44,6 +44,17 @@ def load_arguments_eh(self, _):
...
+
+    with self.argument_context('eventhubs namespace update') as c:
+        c.argument('key_source', options_list=['--key-source'], arg_type=get_enum_type(KeySource), help='A boolean value that indicates whether Zone Redundant is enabled for Namespace.')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:55]: It seems the key source should be keyvault or something else. The help message is strange. could you check it again?

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 3]
```diff
@@ -44,6 +44,17 @@ def load_arguments_eh(self, _):
...
+        c.argument('key_properties', options_list=['--key-properties'], nargs='+', help='List of Key Properties [[keyname,keyvaulturi,keyversion],[keyname,keyvaulturi,keyversion]]')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:56]: For key properties, could we expand all properties and group them using arg_group? You can refer to `az storage account update`.

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 4]
```diff
@@ -44,6 +44,17 @@ def load_arguments_eh(self, _):
...
+        c.argument('user_identity', options_list=['--user-identity'], help='when \'none\' provided Managed Identity is disabled or can set user defined Identity')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:57]: '`none` provided' means there is no value or 'none' str is provide?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:57]: You can see the default value is system assigned identity.

Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:57]: its basically for user to disable the Identity

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py (hunk 3 above)
Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py]: expand all properties and grouped

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py (hunk 2 above)
Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py]: corrected the help message

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py (hunk 1 above)
Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py]: changed to --assign-identity

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 5]
```diff
@@ -230,7 +230,7 @@ def default_api_version(self):
         # to have commands show up in the hybrid profile which happens to have the latest
         # API versions
         ResourceType.MGMT_APPSERVICE: '2018-02-01',
-        ResourceType.MGMT_EVENTHUB: '2017-04-01',
+        ResourceType.MGMT_EVENTHUB: '2018-01-01-preview',
```
Comment (Juliehzl) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:233]: For Azure Stack, it also support the new api version, right?

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 6]
```diff
@@ -44,6 +44,22 @@ def load_arguments_eh(self, _):
...
+        c.argument('zone_redundant', options_list=['--zone-redundant'], arg_type=get_three_state_flag(),
+                   help='Enabling this property creates a Standard EventHubs Namespace in regions supported availability zones')
+        c.argument('identity', arg_group='Managed Identity', options_list=['--assign-identity'], arg_type=get_three_state_flag(),
+                   help='A boolean value that indicates whether Managed Identity is enabled.')
+
+    with self.argument_context('eventhubs namespace create') as c:
+        c.argument('cluster_arm_id', options_list=['--cluster-arm-id'], help='luster ARM ID of the Namespace')
+
+    with self.argument_context('eventhubs namespace update', arg_group='Managed Identity') as c:
+        c.argument('key_source', options_list=['--key-source'], arg_type=get_enum_type(KeySource),
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:56]: If no specified, which kind of key source will be used in eventhub?

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 7]
```diff
@@ -44,6 +44,22 @@ def load_arguments_eh(self, _):
...
+                   help='Encryption key source. Possible values include: \'Microsoft.KeyVault\'.')
+        c.argument('key_name', help='The name of the KeyVault key.', )
+        c.argument('key_vault_uri', help='The Uri of the KeyVault.')
+        c.argument('key_version',
+                   help='The version of the KeyVault key to use, which will opt out of implicit key rotation. '
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:61]: Please make sure if you have key auto-rotation feature.

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 8]
```diff
@@ -44,6 +44,22 @@ def load_arguments_eh(self, _):
         c.argument('maximum_throughput_units', type=int, help='Upper limit of throughput units when AutoInflate is enabled, vaule should be within 0 to 20 throughput units. ( 0 if AutoInflateEnabled = true)')
         c.argument('default_action', arg_group='networkrule', options_list=['--default-action'], arg_type=get_enum_type(['Allow', 'Deny']),
                    help='Default Action for Network Rule Set.')
+        c.argument('zone_redundant', options_list=['--zone-redundant'], arg_type=get_three_state_flag(),
+                   help='Enabling this property creates a Standard EventHubs Namespace in regions supported availability zones')
+        c.argument('identity', arg_group='Managed Identity', options_list=['--assign-identity'], arg_type=get_three_state_flag(),
+                   help='A boolean value that indicates whether Managed Identity is enabled.')
+
+    with self.argument_context('eventhubs namespace create') as c:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:52]: If this command is only supported after 2018-01-01-preview, please add `min_api` args.

File: src/azure-cli/azure/cli/command_modules/eventhubs/custom.py [hunk 9]
```diff
@@ -68,7 +92,7 @@ def cli_namespace_update(cmd, client, instance, tags=None, sku=None, capacity=No
 
 
 def cli_namespace_list(cmd, client, resource_group_name=None):
-    if cmd.supported_api_version(min_api='2017-04-01'):
+    if cmd.supported_api_version(min_api='2018-01-01-preview'):
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:95]: It seems that there is no need to change this, if list() is already supported from 2017-04-01.

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 5 above)
Comment (v-Ajnava) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py:233]: yes

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py (hunk 6 above)
Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:56]: Microsoft.KeyVault

File: src/azure-cli/azure/cli/command_modules/eventhubs/custom.py (hunk 9 above)
Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:95]: list_by_resource_group() is supported in 2018-01-01 preview version

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py (hunk 7 above)
Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:61]: updated the help message

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py (hunk 8 above)
Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:52]: added min_api

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 10]
```diff
@@ -44,6 +44,21 @@ def load_arguments_eh(self, _):
         c.argument('maximum_throughput_units', type=int, help='Upper limit of throughput units when AutoInflate is enabled, vaule should be within 0 to 20 throughput units. ( 0 if AutoInflateEnabled = true)')
         c.argument('default_action', arg_group='networkrule', options_list=['--default-action'], arg_type=get_enum_type(['Allow', 'Deny']),
                    help='Default Action for Network Rule Set.')
+        c.argument('zone_redundant', options_list=['--zone-redundant'], arg_type=get_three_state_flag(),
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:47]:
```suggestion
        c.argument('zone_redundant', options_list=['--zone-redundant'], is_preview=True, arg_type=get_three_state_flag(),
```

Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py]: added is_preview to all new features

**Pull request merged**